src/keys/bip39.rs

```diff
@@ -43,17 +165,43 @@ pub mod wordlist {
     #[cfg_attr(docsrs, doc(cfg(feature = "bip39-jp")))]
     include!("bip39.jp.rs");
 
-    #[derive(Debug, PartialEq, Eq)]
-    pub enum Error {
-        InvalidEntropyCount(usize),
-        NoSuchWord(String),
-        ChecksumMismatch,
-    }
-
     const fn cs(ent: usize) -> usize {
         ent / 32
     }
 
+    impl<'a> Wordlist<'a> {
+        // TODO: should it be pub?
+        const fn new_unchecked(separator: char, words: &'a [&'a str; 2048]) -> Self {
+            Self { words, separator }
+        }
+
+        pub fn new(separator: char, words: &'a [&'a str; 2048]) -> Result<Self, Error> {
+            // normalize separator char
+            let mut s = String::new();
+            s.push(separator);
+            let s: String = s.nfkd().collect();
+            let mut s_chars = s.chars();
```

Comment:
I think you can simplifiy this code (don't need to collect the iterator return by `nfkd()`. Something like this should already work since `Decompositions<I>` implements an iterator over chars:
```
let mut s_chars = String::from(separator).nfkd();
```

Reply:
Done.